src/containers/PageConstructor/components/ConstructorBlocks/ConstructorBlocks.tsx

```diff
@@ -48,13 +53,15 @@ export const ConstructorBlocks = ({items, shouldRenderBlock}: ConstructorBlocksP
             );
         } else {
             if ('children' in item && item.children) {
-                children = item.children.map(renderer);
+                children = item.children.map(renderer.bind(null, blockId));
             }
 
             itemElement = (
-                <ConstructorItem data={item} key={key}>
-                    {children}
-                </ConstructorItem>
+                <BlockIdContext.Provider value={blockId || key}>
```

Comment:
seems like there
```const blockId = parentId ? `${parentId}_${key}` : key;``` we already have key as fallback value for blockId and can use 
`<BlockIdContext.Provider value={blockId}>`

Reply:
exactly